Mobile v2.4 changelog (#6368)
mattermost/docs: source/deploy/mobile-app-changelog.md

## 2.4.0 Release
- Release Date: May 16, 2023
- Server Versions Supported: Server v7.8.0+ is required. Self-Signed SSL Certificates are not supported unless the user installs the CA certificate on their device.

### Compatibility
 - **Upgrade to server version v7.8.0 or later is required.** Support for server [Extended Support Release](https://docs.mattermost.com/upgrade/extended-support-release.html) (ESR) v7.1.0 has ended and upgrading to server ESR v7.8.0 or later is required. As we innovate and offer newer versions of our mobile apps, we maintain backwards compatibility only with supported server versions. Users who upgrade to the newest mobile apps while being connected to an unsupported server version can be exposed to compatibility issues, which can cause crashes or severe bugs that break core functionality of the app.
 - Android operating system 7+ [is required by Google](https://android-developers.googleblog.com/2017/12/improving-app-security-and-performance.html).	
 - iPhone 5s devices and later with iOS 12.1+ is required.

### Improvements
 - Added mobile support for message acknowledgements and persistent notifications.
 - Added the "Add members" screen.

>  - Increased the character limit in the nickname field in the user interface from 22 to 64 characters.

Reviewer: ```suggestion
 - Increased the character limit for the user's nickname from 22 to 64 characters.
```

 - Added localization support for iOS usage description strings in permission dialogs.

>  - Users can now follow or unfollow a thread and undo the action if needed.

Author: @cwarnermm - https://github.com/mattermost/mattermost-mobile/pull/7267

 - Added a channel files screen.

>  - Added support for ``ExperimentalSettings.DelayChannelAutocomplete``.

Author: Yes, server PR: https://github.com/mattermost/mattermost-server/pull/22952 & docs PR: https://github.com/mattermost/docs/pull/6346.

>  - Added support for ``ExperimentalSettings.DelayChannelAutocomplete``.

Reviewer: Thanks!

 - Android operating system 7+ [is required by Google](https://android-developers.googleblog.com/2017/12/improving-app-security-and-performance.html).	
 - iPhone 5s devices and later with iOS 12.1+ is required.

### Improvements
 - Added mobile support for message acknowledgements and persistent notifications.
 - Added the "Add members" screen.
 - Increased the character limit in the nickname field in the user interface from 22 to 64 characters.
 - Added localization support for iOS usage description strings in permission dialogs.
 - Users can now follow or unfollow a thread and undo the action if needed.
 - Added a channel files screen.
 - Added support for ``ExperimentalSettings.DelayChannelAutocomplete``.
 - Calls: Call threads are now auto-followed when joining from mobile.
 - Calls, Android: Fixed an issue with bluetooth; added audio device selection menu.
 - Added support for self-hosted Sentry URL parameters in build scripts.

>  - Added a ``files_count`` column to the ``ChannelInfo`` table and database migration.

Reviewer 2: This one is more or less part of the effort on line 27: `Added a channel files screen.`. 0/5 on whether we should delete this line and just keep the one on line 27.

Author: Thanks, I removed line 32.

### Bug Fixes
 - Fixed an issue where the app did not now show an error when adding servers without a diagnostic ID.
 - Fixed an issue where users were unable to select several files when attaching files to a message.
 - Fixed an issue where some channels would appear as if no messages were there.
 - Fixed an issue with using the camera to capture photos/videos on Android versions 9 and below.
 - Fixed a crash on iOS when attempting to open a Thread by tapping on the notification.

>  - Fixed an issue where channels would appear as empty.

Reviewer 2: This is the same issue as the one on line 37 "Fixed an issue where some channels would appear as if no messages were there.", but needed two different commits. We can delete any of the two lines. 0/5 on which one is more informative.

Author: Thanks, I removed the duplicate one.
